noxfile.py

```diff
@@ -379,6 +383,98 @@ def check_bundled_app(session: nox.Session) -> None:
             )
 
 
+@nox.session()
+def html2jpeg(session: nox.Session):
+    """Create JPEGs from HTML for documentation"""
+
+    port = 12305
+
+    def exporter_running():
+        with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as sock:
+            if sock.connect_ex(("localhost", port)) == 0:
+                return True
+            return False
+
+    if not exporter_running():
+        session.error(
+            f"HTML exporter not running. Please run 'docker run  -p {port}:2305 bedrockio/export-html'"
+        )
```

Comment:
How quickly does the container boot up? If it's not too expensive, we could start the container automatically, figure out it's IP address, and then send requests its way. This would also avoid the need to hardcode a port number.

Something like:

```python
with ExitStack() as defer:
    container_id = subprocess.check_output(["docker", "run", "--rm", "--detach", "IMAGE"])
    defer.callback(subprocess.run, ["docker", "stop", container_id])
    container_config = json.loads(subprocess.check_output(["docker", "inspect", container_id]))
    container_ip = container_config[0]["NetworkSettings"]["IPAddress"]
    url = f"http://{container_ip}:2305/1/screenshot"

    ...
```

Reply:
I thought also about this but that won't work inside the container without DIND. I'll try to modify the session to start the container for doing the conversion and stop it afterwards.

Reply:
For me getting the IP address of the container is working but the container isn't accessible. I changed to get a free port from the OS and use this port to start the docker container.